umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/get_umm_element.clj

```diff
@@ -10,8 +10,9 @@
   "Get collection progress value of UMM based on coll-progress-mapping and xml-path
    If the upper case of the xml-value at xml-path is mapped to a value in coll-progress-mapping,
    that value will be used for collection progress. Otherwise, use \"NOT PROVIDED\""
-  [coll-progress-mapping doc xml-path]
+  [coll-progress-mapping doc xml-path sanitize?]
   (get coll-progress-mapping
        (when-let [xml-value (value-of doc xml-path)]
          (string/upper-case xml-value))
-       umm-spec-util/NOT-PROVIDED))
+       (when sanitize?
```

Comment:
I'm really not sure if this is gonna work. without the testing I can't tell for sure. but I know if NOT-PROVIDED is assigned here, for whatever reason,  when validated, the warning is not gonna be generated.

Reply:
If the collection contains "NOT-PROVIDED", then the schema is not going to see anything missing and won't throw warnings.
(let [{:keys [format metadata]} collection-concept
        collection (spec/parse-metadata context :collection format metadata {:sanitize? false})
        sanitized-collection (spec/parse-metadata context :collection format metadata)
        ;; Throw errors for validation on sanitized collection
        _ (v/umm-spec-validate-collection sanitized-collection validation-options context false)
        ;; Return warnings for schema validation errors going from xml -> UMM
        warnings (v/validate-collection-umm-spec-schema collection validation-options)